### PR TITLE
New build with scikit

### PR DESCRIPTION
**Pull request by repagh**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Please do not merge yet. This version is for testing, when working, will update the version number on Slycot, and reset url to grab from main Slycot development at github.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,34 +1,47 @@
...
-    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}  # [not win]
+    - flang                      # [win]
+    - {{ compiler('c') }}        # [not osx]
+    - gcc                        # [osx]
 
   host:
-    - libblas
-    - liblapack
+    - mkl                        # [linux]
```
Comment (moorepants) [inline recipe/meta.yaml]: I would suggest that mkl somehow be an option for the user. I personally wouldn't want to require this.

Comment (repagh) [reply recipe/meta.yaml]: Hm. 

How can a user select that when installing from conda-forge?
And then we need to add/build more variants.

With the osx build I hit a wall trying to use clang; I reverted to gcc instead, that at least works. I found some vague bug reports on missing development libraries, which is what gets reported when trying clang. The osx script seems not to work with mkl.

The windows and Linux scripts install mkl, also when trying to build other (openblas) variants. I guess these are pulled in by numpy then?

Comment (moorepants) [reply recipe/meta.yaml]: They way it works on linux systems is this: https://stackoverflow.com/questions/33873211/is-it-possible-to-switch-between-blas-libraries-without-recompiling-program

I'm not sure how conda forge is handling this now. Everything used to be all based on openblas.

Comment (repagh) [reply recipe/meta.yaml]: Looks like you are right. I see that the mkl stuff is from anaconda, and conda-forge seems to stick to openblas. 
I converted the meta.yaml to use openblas for all platforms. 

I Also squashed the build modifications in the Slycot PR. Can you merge that and tag 0.3.4? After that I can update the url for this PR, squash, and we can finally have a windows conda-forge package too.

Comment (moorepants) [reply recipe/meta.yaml]: > I Also squashed the build modifications in the Slycot PR. Can you merge that and tag 0.3.4? After that I can update the slycot source url in the meta.yaml, squash, and we can finally have a windows conda-forge package too.

I don't think I have merge permissions on Slycot.

Comment (moorepants) [reply recipe/meta.yaml]: I don't.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
+# * the mkl library does not link on osx or win. Alternative: openblas
 
 requirements:
   build:
-    - {{ compiler('fortran') }}
-    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}  # [not win]
+    - flang                      # [win]
+    - {{ compiler('c') }}        # [not osx]
+    - gcc                        # [osx]
```
Comment (moorepants) [inline recipe/meta.yaml]: Why is gcc used on osx  instead of the default clang compiler?

Comment (moorepants) [reply recipe/meta.yaml]: Nevermind, I see your note above.

**Comment by repagh** [comment]
At least this works. 
If I understand correctly, you want to wait for the pypi source to be updated & build off from there?
If that is final, I will try to squash to two commits, one for the manual changes, the other for the smithy rerender.

**Comment by moorepants** [comment]
Yes, ideally we source from pypi as a best practice and to facilitate the auto prs to this repo on pypi version updates.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
+  # for testing still, should switch over to pypi package when available
+  url: https://github.com/python-control/Slycot/archive/v{{ version }}.tar.gz
+  # url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
   sha256: {{ sha256 }}
 
 build:
-  number: 1003
-  # NOTE: Conda forge lacks openblas on Windows, so skip the Windows build.
-  skip: true  # [win]
+  number: 0
```
Comment (moorepants) [inline recipe/meta.yaml]: This may need to be 1000, not 0.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,35 +1,47 @@
...
+  # problem with python2.7 configuration on windows and osx
+  skip: true                     # [win and py==27]
+  skip: true                     # [osx and py==27]
```
Comment (moorepants) [inline recipe/meta.yaml]: Is it not possible to keep an osx for py27. It was supported before, no?

File: recipe/meta.yaml (hunk 3 above)
Comment (repagh) [reply recipe/meta.yaml]: fine with me, but why does it then say reset the build number to 0?

File: recipe/meta.yaml (hunk 4 above)
Comment (repagh) [reply recipe/meta.yaml]: I just tried again. It fails when loading the module. You are welcome to try and fix it, but I don't think, with python2 being phased out, it is worth the trouble.

Comment (moorepants) [reply recipe/meta.yaml]: I may try after this is merged. Looks like it is now pushed to pypi.

File: recipe/meta.yaml (hunk 3 above)
Comment (moorepants) [reply recipe/meta.yaml]: Actually, this is no longer necessary: https://conda-forge.org/docs/maintainer/maintainer_faq.html?highlight=build%20number#mfaq-build-number-1000

The conda forge details change so fast, I can't keep up.

So leave at 0.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,35 +1,47 @@
 {% set name = "slycot" %}
-{% set version = "0.3.3" %}
-{% set sha256 = "d98ec859a398af8a3917a0d62e0724996c8b3e780813f8685410525aed9fbafd" %}
+{% set version = "0.3.4" %}
+{% set sha256 = "3ae6db8df6b99b4417394404a548c6464248fc7031f21008f992e317f91d23f1" %}
```
Comment (moorepants) [inline recipe/meta.yaml]:
```suggestion
{% set sha256 = "63e648b2be8c8ba0322a5fe286bdcf0e8d28a0c12e9ab1510f2a8725fc1283ec" %}
```

That's the pypi hash.

**Comment by moorepants** [comment]
@repagh Thanks for all the hard work on this! Having windows builds is a big +. Merging!

**Comment by isuruf** [comment]
This is not the correct way to handle blas. The previous way was correct. See https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

**Pull request merged**